modules/common-fs2/src/main/scala/com/snowplowanalytics/snowplow/enrich/common/fs2/Environment.scala

```diff
@@ -279,7 +283,27 @@ object Environment {
     if (preparedRemoteAdapters.nonEmpty)
       Clients
         .mkHttp(remoteAdapters.connectionTimeout, remoteAdapters.readTimeout, remoteAdapters.maxConnections, ec)
+        .map(enableRemoteAdapterMetrics(metrics))
         .map(c => (c.some, preparedRemoteAdapters))
     else Resource.pure[F, (Option[Http4sClient[F]], Map[(String, String), RemoteAdapter])]((None, preparedRemoteAdapters))
   }
+
+  def enableRemoteAdapterMetrics[F[_]: Clock: Sync](metrics: Metrics[F])(inner: Http4sClient[F]): Http4sClient[F] =
+    Http4sClient { req =>
+      for {
+        now <- Resource.eval(Clock[F].realTime(MILLISECONDS))
+        resp <- inner.run(req).recoverWith {
+                  case _: TimeoutException =>
+                    Resource.eval(metrics.remoteAdaptersTimeoutCount.map(_ => Response.timeout[F]))
```

Comment:
I think this will increment the timeout count **and** the 5xx count.

Because you catch the exception and replace it with `Response.timeout[F]` which [is a regular response with a 503 status code](https://github.com/http4s/http4s/blob/v0.21.33/core/src/main/scala/org/http4s/Message.scala#L606).  So then `resp.status.responseClass` will match `Status.ServerError` on line 303.

I don't know if this was your intention, but my biggest complaint is **it is not obvious just by looking at the code**.

There are a few possible alternatives, but my favourite is to increment the timeout count and then let the exception bubble up:

```
resp <- inner.run(req).attemptTap {
      case Left(_: TimeoutException) => Resource.eval(metrics.remoteAdaptersTimeoutCount)
      case _ => Resource.pure[F, Unit](())
    }
```


Furthermore, you catch only timeout exceptions, but other exceptions are possible.  For example, if the api returns invalid bytes, or if the TLS handshake fails, then you get a different exception.  Maybe it's worth talking to techops about which metric should be incremented in these cases.  Currently you are not incrementing any metric if this happens.

Reply:
It was my intention but I wasn't semantically sure about it. I'll talk to TechOps about that and update this accordingly. Thanks for the feedback!